### ISSUE TITLE
Pi3B+ certain pages intermittently not responding.

### ISSUE THREAD

**Issue by bloojoop** [FalconChristmas/fpp#1056]
# Please answer the following:
**FPP Version and Hardware**: 
5.x-master-419-gae10a0b6 running on Pi3B+

**Describe the bug and Steps to reproduce**: 
After upgrading past 5.x-master-350.. and letting the pi sit idle for an hour or more, the help about page will not respond.  The FPP settings page also seems to not respond, however, network, multisync, channel inputs, output modifiers, file manager, scheduler all respond immediately with no issue. Channel Outputs will be delayed but will respond after 20-30 seconds.  Once the channel output page responds, then all others will respond as expected.  I also noticed that if I restart FPPD, all pages will work again just fine until the Pi sits idle for more than an hour.  Additionally, I added a very basic schedule but the schedule would not start if the Pi had gotten to this "delayed response" condition.  Once selecting on the channel outputs, and waiting for it to respond, then the schedule fired off immediately.  This behavior did not happen on build 5.x-master-350 or before.  I have tried this Pi as player and remote and same condition.  In the logs around 8/3/21 9:11 am would be an event where I noticed the non-response and got it functioning again by selecting on the channel outpts page and waiting for it to respond. 

**Expected behavior**: 
Expected behavior would be to have all pages respond immediately. 

**Additional context**: 
I also have a Kulp K8pb and this behavior does NOT happen on the K8.  It is also currently running 419.  My Pi originally was 4.6.1 and I did a branch change to master.  Been updating and working with 5.x since around build 200.   This problem did not occur until after I upgraded past build 350.  After the problem started, I updated the OS with FPPOS alpha 1 and the same issue.  I decided to re-image the card with the Pi image.  I left the PI on build 228 from the image for a while and zero issues.  I upgraded to current issue returned.  I tried a second card and second Pi, I removed the Picap and usb sound card just to get it to as plain jane as possible.  So currently bare Pi (I have used 2 as I have several), with fresh image, configure network, update to current.  Push very basic sequence and channel config from my test xlights folder and the problem returns. very repeatable, just have to wait with it being idle for an hour or so.  

**Additional Attachments**
Please include the log files. To create the file:
1. Open the FPP UI
2. Open the the File Manager (Content Setup->File Manager), 
3. Select the "Logs" tab
4. Click "zip" button.
5. Attach to this ticket.

[FPP_Logs_20210803.1458.zip](https://github.com/FalconChristmas/fpp/files/6925091/FPP_Logs_20210803.1458.zip)

**Comment by darylc** [comment]
@bloojoop From your logs this does not look like an FPP issue.  I see several problems.

1. Complaining about low voltage - try powering your PI with an appropriately sized USB plug pack
2. Crashes, maybe related to voltage, but could be related to corruption or bad SD card or bad pi.  If voltage doesn't help, re-image, if that doesn't help, try another SD card etc.

**Comment by bloojoop** [comment]
Thanks Daryl,  I’m not home till tomorrow. I looked through logs but didn’t see the low voltage issue. I have tried 2 different pi’s and two different cards ( i accidentally physically damaged my old card). But still used same power supply. I’ll look at it but i think it is a standard pi power supply but will check it and i have others. All my testing was with same power supply. Interestingly i never had this issue till i upgrades past 350. Anyway. I’ll report back after swapping the power supply.

**Comment by bloojoop** [comment]
@darylc , So I swapped power supplies tonight.  I noticed the low voltage messages in /var/log/syslog.  After swapping power supplies, around 20:00 8/4.  I started my test sequence.  I let it run until 22:00.  I just let the pi sit there.  Around midnight 00:00 I jumped back on it and the same behavior was happening.  Trying to open help->about page just sat there, status line in browser stated "waiting for picap"... Once I go to the Channel output page and wait for it to respond, then all pages respond just fine.  I attached new logs and added the syslog.1 log file to the bundle since it just rotated.  I find no more low voltage messages after changing power supplies.  The last one indicated in syslog was marked at 8/4 21:58 which was really 15:58.  It appeared that syslog was showing in UTC but after shutdown and swapping power supply it was in local time??  you can  I notice in fppd.log at 22:23 it stated "2021-08-04 22:23:40.085 (3862) [General] /opt/fpp/src/fppd.cpp:226: Crash handler called:  11".  followed by a lot of other logging messages.   This behavior never occurred for me on build 5.x-master-350 or earlier.  Only after upgrading to like 357 or so... Also did notice some errors about undefined index and variables in apache2-error.log.  May not be related.   Earlier today I did upgrade to 5.0 release via fppos update then switched back to master branch and currently on 5.x-master-430.  Again this is the 2nd PI and 2nd card.  This card was imaged with 5.0-alpha, upgraded via git hub, then upgraded via fppos, then switched to master over the last couple of days.  I'm happy to try anything else you may suggest.  I may go ahead and just image  from 5.0 release image and leave it to see if still happens.  

[FPP_Logs_20210805.0508.zip](https://github.com/FalconChristmas/fpp/files/6936058/FPP_Logs_20210805.0508.zip)

**Comment by darylc** [comment]
All you can really do is rule things out one by one.  You can try reimaging with 5.0, I'm not thinking that will help but theres no harm trying.  It feels more like a hardware issue to me (possibly even pi, one of my pi's that worked fine last year has stopped booting now and displays errors)

**Comment by bloojoop** [comment]
I tend to agree but i tried two of my 3b+’s. And the pi seems to be running fine. Just fppd seems to be hung up for a bit. I rolled back to 350 and had no issues for a day. I now swapped the card to a 3b from another project and going to see how it does over night. I’m fairly confident if this was wide spread others would be seeing it so not sure what is unique here. Anyway going to let it run tonight and head to bed.

**Comment by patdelaney** [comment]
Are there any hats installed on your PI that is having the issue, or do you have any plugins installed/configured?  Just looking to try to replicate your environment reproduce your issue here in the lab.

**Comment by bloojoop** [comment]
@all, now running on my pi 3B (no plus) and still same issue ?? @patdelaney , not anymore. I usually have a pi hat and usb sound card on this Pi. I removed them in this trouble shooting.  Current state is, imaged card with 5.x-alpha one.  Updated to current which was like build 405 at the time or close to it.  pushed small config from xlights 2021.23.  Just one enabled output to a WLED controller and a small sequence with 30 secs of butterfly on two props.  One of the props is on my WLED controller and one on my k8-pb running 5.x-master-430 remote.  I do not have this issue on the k8-pb.  In troubleshooting I then upgraded to the 5.0 release with fppos.  Then switched to master and updated to current (right now 5.x-master 430).  Nothing special or interesting about it's current configuration. I am using build in wlan0 only and configured for my IOT network using dhcp.  When this condition is present, the pi itself is responding as normal, I can ping it, ssh into it, collect logs, and browse to all pages except help about and fpp backup, then channel output would be delayed response but always responds after a short bit.
Again the issue is, After the pi just sits for like a hour or more without touching it, Then when I go back to it, the help about page, fpp backup page become completely non-responsive.  Also I noticed in this condition no schedules will run.  All other pages are working just fine as expected.  One interesting note is that I added a schedule and saved it, it showed on the schedule page but would not show in schedule preview.  almost like fppd is stopped or not responding but it does NOT show stopped.  When this happens, the channel output page will act like not responding for 20-30 seconds, but then it will respond just fine.  Once this happens then everything works again, kind of like fppd just got "restarted".  If I do this when a schedule is supposed to be running, it will start right then.  I will also add that when it is stuck, if I just restart fppd, then everything is fine again. So this has now been on 3 different Pi's and two different cards with like 3 different re-images.  This issue did NOT happen until I upgraded newer than 5.x-master-350....  If I roll back to 350, this does not occur.  I am using a HP branded 8GB SLC high end card (i used to use these cards to install and run esxi).  Following DarylC's though that maybe card coruption (I dont really think so since everything work if I keep playing with it, just after is rests for quite a while). I plan to burn 5.0 release onto a new 16GB card and configure for wifi and let it sit with zero configuration other than expand file system and network config with wifi. 

I know this is long but I'm trying to be as detailed as I can. I would be happy to setup a zoom session with anyone that would want to look at it in person or even ship it out to someone that would like to look at it in person :)

**Comment by patdelaney** [comment]
What if you don't push a config from xlights and just let the unit sit?
Does it still have issues?
If it behaves, then just send the inputs and outputs from xLighs but don't
push a sequence and let it run for a while. Does it get hung up?
If not now push the sequence does it get hung up?Are the Controllers online
and active?( WLED and k8-pb)

**Comment by bloojoop** [comment]
I have a little more information and actually have 2 pi's up and running since I have extra :)  I burned a new image and let it sit for 2 hours and no issue at all.  I pushed from xlights and issue came back after sitting for about 1hr 15 min.  Is there an easy way to "revert to factory default" without re-imaging the card?  so likely something that is getting pushed from xlights is causing this behavior.  I could zip my show folder and provide it if needed.  It is small and only one of my test folders.  It is likely about 30 minutes from hanging up again so I will let it sit and then I can manually delete models, sequence and outputs but no guarantee that would be back to "factory default".  Attached is the fpp connect config I used to push to this fresh image. 

![fpp connect](https://user-images.githubusercontent.com/88390168/128411499-2c333b32-e196-416e-b6a2-50ae757e6efb.png)

**Comment by patdelaney** [comment]
So that is exactly where I was going.  The stock image is fine, but after
you push a sequence via FPP Connect.  For clarification. Did you upload
inputs and upload outputs from the controllers tab first?[image: image.png]

Pat

**Comment by bloojoop** [comment]
no.. this pi is not functioning as a controller.  at this time.  I have removed the pi hat and do NOT have the pi pixel stings enabled.  So after base stock image, was fine.  I pushed fpp config just as I shared above and then after sitting 1 hr 15 min with nothing happening, it started to do the same thing where fppd seems to be hung up for a bit and the crash handler log gets written to fppd.log.

**Comment by dkulp** [comment]
To revert to a "stock" setup, you can ssh in and do a "sudo rm -rf /home/fpp/media/*" and then reboot.

**Comment by bloojoop** [comment]
Thanks @dkulp !!  .  attached is a syslog snipit from what appears to be just before the fppd crash.  I see a message about failure to resolve dns name of wled-pixnode1.  That controller is defined but inactive in xlights.  It is indeed offline and not resolvable.  I do use DNS and DHCP for this environment and has been working extremely well for me for about 3 years now.  Anyway, This controller is defined in xlights as inactive but fpp connect still pushes it to the FPP and it shows up in the outputs but is not enabled.  I'm surprised it is trying to resolve a hostname for an inactive and disabled output.  Anyway, this may or may not be related.  Just an additional observation. 

[syslog snipit.txt](https://github.com/FalconChristmas/fpp/files/6941433/syslog.snipit.txt)

**Comment by bloojoop** [comment]
I removed the disabled output for that offline controller and will see if that makes a difference.

**Comment by bloojoop** [comment]
OK, here is my latest results.  I have 2 pi's running fpp, one on 5.0.1 (named picap) and one on 5.x-master-430- (named testpi1).  I cleared the config so they are basically out of the box.  No issues and left them running for 2 hours.  I then pushed the upd all output only using xlights fpp connect to both instances.  after 1 hr and 17 minutes both had a crash handler errors in the fppd.log. This is the sign that something is wrong.  I then rolled testpi1 (the one on master) back to version 350 from 3 weeks ago (right before the player/remote mode changes).  I then let both Pi's stand idle with the same output configuration.  The 5.0.1 picap crashed again after 1 hr 19 mins.  The 5.x-master-350 testpi1 did not crash even after 2 hours.  It has been fine with the exact same output configuration.  I have logs for both uploaded to this message.  it is obvious that there is something in the code changes since 350 that is causing this with my output configuration.  It is also interesting that this occurs very consistently at 1 hr 17-19 mins after fppd restart. 

[FPP_Logs_20210806.0458_ffpderrors.zip](https://github.com/FalconChristmas/fpp/files/6943112/FPP_Logs_20210806.0458_ffpderrors.zip)
[FPP_Logs_20210806.0459_build350noerrors.zip](https://github.com/FalconChristmas/fpp/files/6943113/FPP_Logs_20210806.0459_build350noerrors.zip)

**Comment by dkulp** [comment]
Can you pull latest master and test again.   I THINK the crash at ~80 minutes would be gone, but not sure if the other issues are addressed by that or not.

**Comment by bloojoop** [comment]
@dkulp, I updated to master-435 on two of my devices, one Pi and my k8pb.  I waited 2 hours.  No fppd crash and everything appeared to work fine for me !!  YAY. I have reconfigured all for my standard testing and added a schedule so I will report if I run into any further issues but this build looks promising!  extra donation sent ;) I really appreciate the work you all are doing! 

I will add that I had zero issues on the k8pb through all of this. Of course it was not configured with any udp output.  I did add one upd output and then it did experience the problem.  I see your fix was in multisync.cpp and just wanted to report that the trigger in my case was existence of an enabled upd output (I usually use ddp-onebased but I tried e1.31 as well same issue).

**Comment by darylc** [comment]
@bloojoop Just wanted to thank you for your excellent testing.

**Comment by bloojoop** [comment]
> @bloojoop Just wanted to thank you for your excellent testing.

@darylc, you're welcome, I'm always glad to contribute and help where I can.

**Comment by dkulp** [comment]
The main issue is fixed, if there are additional issues, please log new bugs.  Thanks!
